src/System.IO.Pipelines.Networking.Tls/ISecureContext.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System.Threading.Tasks;
+
+namespace System.IO.Pipelines.Networking.Tls
+{
+    public interface ISecureContext : IDisposable
```

Comment:
has the side effect that SecurePipeline now becomes internal due to the <T>. Not sure how I feel about this, if anyone wants to extend this lib outside the main library we just lost an extension point
rewriting the SecurePipeline seems silly for each external library as it is very generic and OpenSsl/Sspi etc agnostic..

Thoughts?